db/compaction_job.cc

```diff
@@ -1137,8 +1138,8 @@ Status CompactionJob::FinishCompactionOutputFile(
     // we will regrad this verification as user reads since the goal is
     // to cache it here for further user reads
     InternalIterator* iter = cfd->table_cache()->NewIterator(
-        ReadOptions(), env_options_, cfd->internal_comparator(), meta->fd,
-        nullptr /* range_del_agg */, nullptr,
+        ReadOptions(), env_options_for_write_, cfd->internal_comparator(),
```

Comment:
should this be env_options_for_read_?

Reply:
we can find a better name for `CompactionJob:: env_options_for_write_ ` but here we should use the original env_options_, not `CompactionJob::env_options_for_read_`